### DISCUSSION TITLE
cfde-submit submission and frictionless validator error related to JSON

### DISCUSSION THREAD

**Discussion by ericwenger-pm** [nih-cfde/published-documentation#303] [Q&A] [answered]
cfde-submit fails with error "Validation error in The task has an error: 'fieldPosition'"

**Details**
Is there an updated version of the C2M2_datapackage.json file?  Or is there a resolution for the issue outlined below with the .JSON file failing validation.  I encountered this error also when pre-validating all of the .tsv files along with the JSON, after running the helper script when running frictionless validator.  Only the JSON did not pass validation.  Output of the validation attached below:
[~frictionlessreport_c2m2_submission_year2_v3.txt](https://github.com/nih-cfde/published-documentation/files/7832168/frictionlessreport_c2m2_submission_year2_v3.txt)

The version of JSON I used is the most recent of 2021-11-02 03:49 PM (version 3) from https://osf.io/vzgx9/ 

Here is the cfde-submit error message
```
(cfde_venv_year2_submit) 909c4acc5cca:~ wengere$ cfde-submit run ~/c2m2/c2m2_submission_year2_baseline_test_v1_3
Using saved DCC 'cfde_registry_dcc:kidsfirst'
Starting login with Globus Auth, press ^C twice to cancel or once to manually authenticate
You are authenticated and your tokens have been cached.
Submit datapackage 'c2m2_submission_year2_baseline_test_v1_3' using cfde_registry_dcc:kidsfirst? (y/n)? > y
Validation error in The task has an error: 'fieldPosition'
(cfde_venv_year2_submit) 909c4acc5cca:~ wengere$ 
```

Here is the code I used to try to get more detailed error details, since the initial output was limited:
```
from pprint import pprint
from frictionless import validate
from frictionless import validate_schema

report = validate_schema('C2M2_datapackage.json')
pprint(report)
```

output from above
```
(virtualenvY2Q1_c2m2helper) 909c4acc5cca:frictionless_validate_combined_output wengere$ python validate_json_report.py
{'errors': [{'code': 'schema-error',
             'description': 'Provided schema is not valid.',
             'message': 'Schema is not valid: "\'fields\' is a required '
                        'property" at "" in metadata and at "required" in '
                        'profile',
             'name': 'Schema Error',
             'note': '"\'fields\' is a required property" at "" in metadata '
                     'and at "required" in profile',
             'tags': []}],
 'stats': {'errors': 1, 'tasks': 0},
 'tasks': [],
 'time': 0.001,
 'valid': False,
 'version': '4.23.0'}
```

I also tested this against the earlier versions of the JSON (September 2021 version, for example) and the earlier version used in our last successful submission, and the same error occurs.  Was not sure whether this is a frictionless issue (maybe similar to https://github.com/frictionlessdata/frictionless-py/issues/838 ) but with additional testing could not eliminate the error.

**Comment by ericwenger-pm** [comment, marked as answer]
Issue resolved

Identified the issue: my version of the dcc.tsv that I created was missing the new id column.  I corrected this, and also put the order of the columns in the same order as the fields defined in the .JSON. 

After doing this, I re-ran the helper script and then re-reran the frictionless validator.  The JSON error was resolved and all files were valid.  Output attached below.
[~frictionlessreport_c2m2_submission_year2_v4.txt](https://github.com/nih-cfde/published-documentation/files/7837684/frictionlessreport_c2m2_submission_year2_v4.txt)

**Reply by ACharbonneau** [reply]
Yay! Sorry I didn't see this until this morning. Glad you found it :)
